tutorials/nservicebus-101/lesson-4/tutorial.md

```diff
@@ -0,0 +1,212 @@
+```
+
+That's great, but why stop there? The whole point of Publish/Subscribe is that we can have *multiple* subscribers.
+
+
+### Create another subscriber
+
+In a real system, after an order is placed and billed, we would need to ship the products. So let's add another event and two more subscribers. Once the credit card is charged, we'll publish an `OrderBilled` event. Then, we'll create a new endpoint **Shipping** that will subscribe to both events.
+
+This is also a good opportunity to check your understanding. If you can complete these steps without looking back at previous steps or previous lessons, you can be sure you have a good understanding of everything we've covered so far. (Don't worry, you can always check your work against the solution.)
+
+ 1. In **Messages**, create a new event called `OrderBilled`, implementing `IEvent` and containing a property for the `OrderId`.
+ 1. In **Billing**, publish the `OrderBilled` event at the end of the `OrderPlacedHandler`.
+ 1. Create a new endpoint named **Shipping** with the necessary dependencies. Be sure to set the console title and endpoint name to `"Shipping"`, and configure it to start when debugging.
+ 1. In **Shipping**, create a message handler for `OrderPlaced`.
+ 1. In **Shipping**, create a message handler for `OrderBilled`.
+ 1. Configure **Shipping** to subscribe to `OrderPlaced` from **Sales**. (You may have this already if you copied the endpoint configuration from **Billing**.)
+ 1. Configure **Shipping** to subscribe to `OrderBilled` from **Billing**. (This will be new no matter what.)
+
+
+### Running the solution
+
+If everything worked, you should now see output like this in your **Shipping** window:
+
+```no-highlight
+INFO  Shipping.OrderPlacedHandler Received OrderPlaced, OrderId = 96ee660a-5dd7-4772-9058-863d303ee0aa - Should we ship now?
+INFO  Shipping.OrderBilledHandler Received OrderBilled, OrderId = 96ee660a-5dd7-4772-9058-863d303ee0aa - Should we ship now?
+```
+
+Of course, these messages could appear out of order. With asynchronous messaging, there are no message ordering guarantees. Even though `OrderBilled` comes logically after `OrderPlaced`, it's possible that `OrderBilled` could arrive first.
+
+You'll note that in the sample solution, the message for each handler says "Should we ship now?" This is because both message handlers are stateless. Like HTTP requests, message handlers have no intrinsic memory of what came before. In an upcoming course, we'll explore [Sagas](/nservicebus/sagas/), an NServiceBus feature that provides that memory in-between message handlers, similar to how ASP.NET Session State stores data between HTTP requests in ASP.NET web applications. Once we have that capability, we'll be able to publish an `OrderShipped` event once both the `OrderPlaced` and `OrderBilled` events arrive, but this is good enough for now.
```

Comment:
consider the many negative connotations of asp sessions satte i think we should drop this 

> similar to how ASP.NET Session State stores data between HTTP requests in ASP.NET web applications